public/electron.js

```diff
@@ -33,8 +34,65 @@ function createWindow() {
     mainWindow.show();
   });
   ipcHandler(mainWindow);
+
+  mainWindow.on("minimize", (e) => {
+    e.preventDefault();
+    mainWindow.hide();
+  });
+
+  mainWindow.on("close", (e) => {
+    if (!app.isQuiting) {
+      e.preventDefault();
+      mainWindow.hide();
+      tray.setContextMenu(trayMenuWithShow);
+    }
+  });
 }
 
+const trayMenuWithShow = Menu.buildFromTemplate([
+  {
+    label: "Show",
+    click: () => {
+      mainWindow.show();
+      tray.setContextMenu(trayMenuWithHide);
+    },
+  },
+  {
+    label: "Quit",
+    click: () => {
+      app.isQuiting = true;
+      app.quit();
+    },
+  },
+]);
+
+const trayMenuWithHide = Menu.buildFromTemplate([
+  {
+    label: "Hide",
+    click: () => {
+      mainWindow.hide();
+      tray.setContextMenu(trayMenuWithShow);
+    },
+  },
+  {
+    label: "Quit",
+    click: () => {
+      app.isQuiting = true;
+      app.quit();
+    },
+  },
+]);
+
+app
+  .whenReady()
+  .then(() => {
+    tray = new Tray(path.join(__dirname, "./assets/512x512.png"));
+    tray.setContextMenu(trayMenuWithHide);
+  })
+  .catch((e) => {
+    console.log(e);
```

Comment:
I'd suggest logging the error to the log file instead of console:
import `log` from `electron-log` and use it `log.error(e)`.

Reply:
Done.